odoo/addons/base/models/res_currency.py

```diff
@@ -248,26 +275,34 @@ class CurrencyRate(models.Model):
         ('currency_rate_check', 'CHECK (rate>0)', 'The currency rate must be strictly positive.'),
     ]
 
-    def _get_latest_rate(self):
-        return self.search([
-            ('currency_id', '=', self.currency_id.id),
-            ('company_id', '=', self.company_id.id or self.env.company.id),
-            ('name', '<=', self.name or fields.Date.today()),
-        ], order="name desc", limit=1)
+    def _get_latest_rate(self, excluded_name=None):
+        rates = self.currency_id.rate_ids.filtered(
+            lambda x: x.rate and x.company_id == self.company_id and x.name != excluded_name).sorted('name')
+        return rates and rates[-1] or rates
+
+    def _inverse_inverse_rate(self):
+        for currency_rate in self:
+            currency_rate.rate = 1.0 / currency_rate.inverse_rate
 
-    @api.depends('currency_id', 'company_id', 'name')
+    @api.depends('currency_id', 'company_id')
     def _compute_rate(self):
         for currency_rate in self:
-            latest_rate = currency_rate._get_latest_rate()
-            currency_rate.rate = latest_rate.rate if latest_rate else 1.0
+            currency_rate.rate = currency_rate.rate or self._get_latest_rate().rate or 1.0
+
+    @api.depends('rate')
+    def _compute_inverse_rate(self):
+        for currency_rate in self:
+            currency_rate.inverse_rate = 1.0 / (currency_rate.rate or 1.0)
+
+    @api.onchange('inverse_rate')
+    def _on_change_inverse_rate(self):
+        # force inverse _inverse_inverse_rate to update view
+        self._inverse_inverse_rate()
```

Comment:
Doesn't just putting the decorator on `_inverse_inverse_rate` work?

Reply:
Yes, indeed.